crates/oxc_linter/src/rules/eslint/no_throw_literal.rs

```diff
@@ -0,0 +1,215 @@
+use oxc_ast::{
+    ast::{AssignmentOperator, Expression, LogicalOperator},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_throw_literal_diagnostic(span: Span, is_undef: bool) -> OxcDiagnostic {
+    let message =
+        if is_undef { "Do not throw undefined" } else { "Expected an error object to be thrown" };
+
+    OxcDiagnostic::warn(message).with_label(span)
```

Comment:
Could we add help text for this? Like:

- ```
  Wrap the value with an error object: `throw new Error(...)`
  ```
- or if `undefined`:
  ```
  throw an error object instead: `throw new Error()`
  ```

Reply:
Currently, there are many rules that do not provide help, only help prompts when fixable.

> Wrap the value with an error object: `throw new Error(...)`

For fixable types like `string` and `template`, they automatically include help suggestions. We've already implemented fixes for some of them, and the remaining ones are largely unfixable, so this prompt might not be necessary.

> throw an error object instead: `throw new Error()`

For `undefined`, this should be a very intuitive error prompt in itself, and perhaps `help` is not necessary.


Is there a necessary requirement for adding `help`? Under what circumstances should we add it? @DonIsaac

Reply:
consider this comment as non-blocking. I'm of the opinion that every warning/error should include at least _some_ indication of how to fix it. I think a static message here would be fine too:

```rust
.with_help("Throw an Error object instead: `throw new Error()`")
```

Some diagnostics don't contain help messages, I would consider that something we should fix. From our [Linter Rule principles](https://oxc.rs/docs/contribute/linter/adding-rules.html#diagnostics):

> - The `message` should be an imperative statement about what is wrong, not a description of what the rule does.
> - The `help` message should be a command-like statement that tells the user how to fix the issue.